phpunit

```diff
@@ -0,0 +1,124 @@
+#!/usr/bin/env php
+<?php
+
+error_reporting(-1);
+
+$PHPUNIT_VERSION = 4.8;
+$PHPUNIT_DIR = realpath(dirname($argv[0])).'/.phpunit';
+
+if (!file_exists("$PHPUNIT_DIR/phpunit-$PHPUNIT_VERSION/phpunit")) {
+    # Build a standalone phpunit without symfony/yaml
+
+    $PHPUNIT_PWD_BACKUP = getcwd();
+    mkdir($PHPUNIT_DIR);
+    chdir($PHPUNIT_DIR);
+    if (extension_loaded('openssl') && ini_get('allow_url_fopen')) {
+        stream_copy_to_stream(fopen("https://github.com/sebastianbergmann/phpunit/archive/$PHPUNIT_VERSION.zip", 'rb'), fopen("$PHPUNIT_VERSION.zip", 'wb'));
+    } else {
+        passthru("wget https://github.com/sebastianbergmann/phpunit/archive/$PHPUNIT_VERSION.zip");
+    }
+    $zip = new ZipArchive();
+    $zip->open("$PHPUNIT_VERSION.zip");
+    $zip->extractTo(getcwd());
+    $zip->close();
+    chdir("phpunit-$PHPUNIT_VERSION");
+    passthru("composer remove --no-update symfony/yaml");
+    passthru("composer install --prefer-source --no-progress --ansi");
+    chdir($PHPUNIT_PWD_BACKUP);
+}
+
+$cmd = array_map('escapeshellarg', $argv);
+$exit = 0;
+
+if (isset($argv[1]) && 'symfony' === $argv[1]) {
+    # Find Symfony components in plain php for Windows portability
+
+    $finder = new RecursiveDirectoryIterator('src/Symfony', FilesystemIterator::KEY_AS_FILENAME | FilesystemIterator::UNIX_PATHS);
+    $finder = new RecursiveIteratorIterator($finder);
+    $finder->setMaxDepth(3);
+
+    array_shift($cmd);
+    $cmd[0] = "php $PHPUNIT_DIR/phpunit-$PHPUNIT_VERSION/phpunit --colors=always";
+    $procs = array();
+
+    foreach ($finder as $k => $v) {
+        if ('phpunit.xml.dist' === $k) {
+            $component = dirname($v->getPathname());
+
+            # Run phpunit tests in parallel
+
+            $c = escapeshellarg($component);
+
+            if ($p = proc_open(implode(' ', $cmd)." $c > $c/phpunit.stdout 2> $c/phpunit.stderr", array(), $pipes)) {
+                $procs[$component] = $p;
+            } else {
+                $exit = 1;
+                echo "\033[41mKO\033[0m $component\n\n";
+            }
+        }
+    }
+
+    foreach ($procs as $component => $p) {
+        do {
+            usleep(300000);
+            $s = proc_get_status($p);
+        } while ($s['running']);
+
+        proc_close($p);
+
+        foreach (array('out', 'err') as $p) {
+            $p = "$component/phpunit.std$p";
```

Comment:
Can you avoid reusing the same variable name for many different purposes ? It makes the code less readable.
And using meaningful variables is also better than using single-letter variables